pyccel/ast/numpyext.py

```diff
@@ -561,9 +575,10 @@ def __init__(self, arg, dtype=None, order='C'):
             if order not in ('K', 'A', 'C', 'F'):
                 raise ValueError(f"Cannot recognize '{order}' order")
 
-            # TODO [YG, 18.02.2020]: set correct order based on input array
             if order in ('K', 'A'):
-                order = 'C'
+                order = arg.order
+                if order is None:
+                    order = 'C'
```

Comment:
A more Pythonic one-liner:
```suggestion
                order = arg.order or 'C'
```